node/src/accountant/payment_adjuster/non_unit_tests/mod.rs

```diff
@@ -497,18 +558,57 @@ fn administrate_single_scenario_result(
     ScenarioResult::new(reinterpreted_result)
 }
 
-fn interpretable_adjustment_results(
-    account_infos: Vec<AccountInfo>,
-    adjusted_accounts: &mut Vec<PayableAccount>,
-) -> Vec<InterpretableAdjustmentResult> {
-    account_infos
+fn merge_information_about_particular_account(
+    accounts_infos: Vec<AccountInfo>,
+    accounts_after_adjustment: Vec<PayableAccount>,
+) -> Vec<(AccountInfo, Option<PayableAccount>)> {
+    let mut accounts_hashmap = accounts_after_adjustment
+        .into_iter()
+        .map(|account| (account.wallet.address(), account))
+        .collect::<HashMap<Address, PayableAccount>>();
+
+    accounts_infos
         .into_iter()
-        .map(|account_info| {
-            prepare_interpretable_adjustment_result(account_info, adjusted_accounts)
+        .map(|info| {
+            let adjusted_account_opt = accounts_hashmap.remove(&info.wallet);
+            (info, adjusted_account_opt)
         })
         .collect()
 }
 
+enum PercentPortionOfCWUsed {
+    Percents(u8),
+    LessThanOnePercent,
+}
+
+impl PercentPortionOfCWUsed {
+    fn new(adjusted_accounts: &[PayableAccount], common: &CommonScenarioInfo) -> Self {
+        let used_absolute: u128 = sum_as(adjusted_accounts, |account| account.balance_wei);
+        let percents = ((100 * used_absolute) / common.cw_service_fee_balance_minor) as u8;
+        if percents >= 1 {
```

Comment:
Maybe it would be better if we reorder that such that if it's `< 1`, then the other enum variant is returned.